tiledb/sm/array/array.cc

```diff
@@ -87,6 +87,7 @@ const EncryptionKey* Array::encryption_key() const {
   return &encryption_key_;
 }
 
+/* This is a deprecated API. */
```

Comment:
Deprecating APIs are only applicable to the public C/C++ APIs. Remove this comment. Same elsewhere.